phoenix-core-server/src/main/java/org/apache/phoenix/iterate/NonAggregateRegionScannerFactory.java

```diff
@@ -465,6 +475,10 @@ public boolean next(List<Cell> results) throws IOException {
                         }
                     }
                     tuple = nextTuple;
+                    if (regionScannerContext != null) {
+                        ScannerContextUtil.updateMetrics(regionScannerContext, scannerContext);
+                        regionScannerContext = null;
+                    }
```

Comment:
I wonder if it is still possible to not pass the ScannerContext to iterators above and still use it here?
Probably we can, since we are not really updating any context in `OrderedResultIterator` and `OffsetResultIterator`? Or i am missing something?

Reply:
@virajjasani how would we pass the context from RegionScannerResultIterator to NonAggregateRegionScannerFactory?

Reply:
Tried to address in latest commit